app/src/main/java/monster/minions/binocularss/activities/SettingsActivity.kt

```diff
@@ -80,11 +80,10 @@ class SettingsActivity : ComponentActivity() {
         theme = sharedPref.getString(THEME, "System Default").toString()
         cacheExpiration = sharedPref.getLong(CACHE_EXPIRATION, 0L)
 
-        db = Room
-            .databaseBuilder(this, AppDatabase::class.java, "feed-db")
-            .allowMainThreadQueries()
-            .build()
-        feedDao = (db as AppDatabase).feedDao()
+        dataGateway = DatabaseGateway()
+        dataGateway.setContext(context = this)
+        dataGateway.setDb()
+        dataGateway.setFeedDao()
```

Comment:
Everything looks very good clean architecture wise but perhaps we could simplify things more by having a constructor that you pass the context, (`this`) to where it effectively calls all of these functions I don't see any reason why that would break things or why these functions would ever need to be called on their own so I think that would be a good way to make a good change even more clean! Good work!

Reply:
Thanks for the suggestion. Will implement these changes asap.